website/docs/setup/guides/advanced-configs.md

```diff
@@ -0,0 +1,203 @@
+---
+title: Developing Complex Vector Config Files
+sidebar_label: Advanced Configs
+description: Strategies for building and maintaining complex Vector configs
+---
+
+Writing large configuration files is not yet an official olympic event. However,
+it's still a good idea to get yourself ahead of the competition. In this guide
+we're going to cover some tips and tricks that will help you write clear, bug
+free Vector configs that are easy to maintain.
+
+## Building Pipelines
+
+In Vector each component of a pipeline specifies which components it consumes
+events from. This makes it very easy to build multiplexed topologies. However,
+hooking up transforms this way can sometimes be frustrating and brittle as the
+number of transforms increases.
+
+Luckily, the Vector team are desperate for your approval and have worked hard to
+mitigate this with the following tools.
+
+### TODO: Talk about compose transform or pipelines
+
+TODO: Once added we can reduce the `generate` section to just a side note, or
+tip.
+
+This saves you the burden of maintaining the links between chained transforms.
+Instead you can simply insert transforms exactly where they need to be within
+the chain, and remove them without breaking the topology.
+
+### Generating Chains
+
+In cases where we simply want to chain a few components together we can avoid
+writing most of the boilerplate with the `generate` subcommand. This allows us
+to express a list of components, where it generates a config with all of those
+components connected in a linear chain.
+
+For example, if we wished to create a chain of three transforms; `json_parser`,
+`add_fields`, and `remove_fields`, we can run:
+
+```bash
+vector generate /json_parser,add_fields,remove_fields > example.toml
+# Find out more with `vector generate --help`
+```
+
+And most of the boilerplate will be written for us, with each component printed
+with an `inputs` field that specifies the component before it:
+
+```toml
+[transforms.transform0]
+  inputs = [ "somewhere" ]
+  type = "json_parser"
+  # etc ...
+
+[transforms.transform1]
+  inputs = [ "transform0" ]
+  type = "add_fields"
+  # etc ...
+
+[transforms.transform2]
+  inputs = [ "transform1" ]
+  type = "remove_fields"
+  # etc ...
+```
+
+The names of the generated components are sequential (`transform0`,
+`transform1`, and so on). It's therefore worth doing a search and replace with
+your editor to give them better names, e.g. `s/transform2/scrub_emails/g`.
+
+### Swimlanes
```

Comment:
This seems a little out of place to me. This is more about building topologies and controlling flow, and less about config development and maintenance. Do you agree?

Reply:
Yeah I'm not attached to it.